strawberry/tools/depth_limit_validator.py

```diff
@@ -0,0 +1,183 @@
+import re
+from typing import Callable, Dict, List, Optional, Union
+
+from graphql import GraphQLError
+from graphql.language import (
+    DefinitionNode,
+    FieldNode,
+    FragmentDefinitionNode,
+    FragmentSpreadNode,
+    InlineFragmentNode,
+    Node,
+    OperationDefinitionNode,
+)
+from graphql.validation import ValidationContext, ValidationRule
+
+from strawberry.extensions.tracing.utils import is_instrospection_key
+
+
+IgnoreType = Union[Callable[[str], bool], re.Pattern, str]
+
+
+def depth_limit_validator(
+    max_depth: int,
+    ignore: Optional[List[IgnoreType]] = None,
+    callback: Callable[[Dict[str, int]], None] = None,
+):
+    """
+    Creates a validator for the GraphQL query depth
+
+    This is a Python port of https://github.com/stems/graphql-depth-limit
```

Comment:
If this is a port, do we need to add the upstream license for this?

Reply:
Potentially the port should even be placed in its own repo?

Reply:
Good call mentioning the upstream license. For reference: graphql-core is a port of graphql.js. Both are licensed under the MIT license (just like strawberry and the original graphql-depth-limit code). graphql-core retained graphql.js' license and copyright notice.

Reply:
Yes good point @BryceBeagle , I'll add the copyright notice. I don't think it needs to be in its own repo though since it's a pretty small amount of code.